dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/pages/security/TokenPage.java

```diff
@@ -0,0 +1,131 @@
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.dolphinscheduler.e2e.pages.security;
+
+import org.apache.dolphinscheduler.e2e.pages.common.NavBarPage;
+import org.apache.dolphinscheduler.e2e.pages.security.SecurityPage.Tab;
+
+import java.util.List;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.FindBys;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import lombok.Getter;
+
+@Getter
+public final class TokenPage extends NavBarPage implements Tab {
+    @FindBy(id = "btnCreateToken")
+    private WebElement buttonCreateToken;
+
+    @FindBy(className = "items")
+    private List<WebElement> tokenList;
+
+    @FindBys({
+        @FindBy(className = "el-popconfirm"),
+        @FindBy(className = "el-button--primary"),
+    })
+    private List<WebElement> buttonConfirm;
+
+    @FindBy(className = "userName")
+    private List<WebElement> userName;
+
+    @FindBy(className = "token")
+    private List<WebElement> token;
+
+    private final TokenForm createTokenForm = new TokenForm();
+    private final TokenForm editTokenForm = new TokenForm();
+
+    public TokenPage(RemoteWebDriver driver) {
+        super(driver);
+    }
+
+    public TokenPage create() {
+        buttonCreateToken().click();
+        createTokenForm().buttonGenerateToken().click();
+        new WebDriverWait(driver, 10).until(ExpectedConditions.elementToBeClickable(createTokenForm.buttonGenerateToken));
+        createTokenForm().buttonSubmit().click();
+        return this;
+    }
+
+    public TokenPage update(String userName) {
+        List<WebElement> tokenList = driver.findElementsByClassName("items");
+        tokenList.stream()
+            .filter(it -> it.findElement(By.className("userName")).getAttribute("innerHTML").contains(userName))
+            .flatMap(it -> it.findElements(By.className("edit")).stream())
+            .filter(WebElement::isDisplayed)
+            .findFirst()
+            .orElseThrow(() -> new RuntimeException("No edit button in token list"))
+            .click();
+
+        TokenForm editTokenForm = new TokenForm();
+
+        editTokenForm.buttonGenerateToken().click();
+        new WebDriverWait(driver, 10).until(ExpectedConditions.elementToBeClickable(createTokenForm.buttonGenerateToken));
+        editTokenForm.buttonSubmit().click();
+        return this;
+    }
+
+    public String getToken() {
+        return driver.findElementByClassName("token").getText();
```

Comment:
It's better we add an argument to method `getToken(String userName)`, to get the token of `userName`, for now it's OK because we only test the case where there is only one token in the list

Reply:
@QuakeWang you added the method argument but it's not used, I added the real logic of the argument, please take a look.